packages/table/src/Table.ts

```diff
@@ -0,0 +1,463 @@
+        `;
+    }
+
+    protected override willUpdate(changed: PropertyValues<this>): void {
+        if (!this.hasUpdated) {
+            const rowValues = new Set<string>();
+
+            if (this.isVirtualized) {
+                this.items.forEach((item, index) => {
+                    const value = this.itemValue(item, index);
+                    rowValues.add(value);
+                });
+            } else {
+                const tableRows = [
+                    ...this.querySelectorAll('sp-table-row'),
+                ] as TableRow[];
+                tableRows.forEach((row) => {
+                    rowValues.add(row.value);
+                });
+            }
+
+            const oldSelectedCount = this.selected.length;
+
+            this.selected = this.selected.filter((selectedItem) =>
+                rowValues.has(selectedItem)
+            );
+            if (oldSelectedCount !== this.selected.length) {
+                this.dispatchEvent(
+                    new Event('change', {
+                        cancelable: true,
+                        bubbles: true,
+                        composed: true,
+                    })
+                );
+            }
+            this.selectedSet = new Set(this.selected);
+
+            this.manageCheckboxes();
+        }
+        if (changed.has('selects')) {
+            this.manageSelects();
+        }
+
+        if (changed.has('selected') && this.hasUpdated) {
+            this.manageSelected();
+        }
+    }
+
+    protected override updated(): void {
+        if (this.items.length) {
+            this.renderVirtualizedItems();
+        }
+    }
+
+    protected renderVirtualizedItems(index?: number): void {
+        if (!this.tableBody) {
+            this.tableBody = this.querySelector('sp-table-body') as TableBody;
+            if (!this.tableBody) {
+                this.tableBody = document.createElement('sp-table-body');
+                this.append(this.tableBody);
+            }
+            // TODO: query Lit team about the `e.stopPropagation()` that happens here.
+            this.tableBody.addEventListener(
+                'rangeChanged',
+                (event: RangeChangedEvent) => {
+                    // console.log('rangeChanged');
+                    this.dispatchEvent(
+                        new RangeChangedEvent({
+                            first: event.first,
+                            last: event.last,
+                        })
+                    );
+                }
+            );
+        }
+        const config: {
+            renderItem?: (
+                item: Record<string, unknown>,
+                index: number
+            ) => TemplateResult;
+            scroller?: boolean;
+            items?: Array<unknown>;
+            scrollToIndex?: {
+                index: number;
+            };
+        } = {
+            items: this.items,
+            renderItem: this.renderItem,
+            scroller: true,
+        };
```

Comment:
Can you look again to see whether this type is exported from the virtualizer? If not, we should create an issue to have it exported for us. We should also touch base on you making some of the smaller changes we would like to see as contributions to their repo.